### PR TITLE
NODE-124: Measure forks frequency and microblock receiving lag

### PR DESCRIPTION
**Pull request by vsuharnikov**

### CODE REVIEW

File: src/main/scala/com/wavesplatform/network/MicroBlockSynchronizer.scala [hunk 1]
```diff
@@ -14,6 +18,8 @@ import scala.concurrent.duration.FiniteDuration
 class MicroBlockSynchronizer(settings: Settings, history: NgHistory)
   extends ChannelInboundHandlerAdapter with ScorexLogging {
 
+  private val microBlockCreationTime = new ConcurrentHashMap[ByteStr, Long]()
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/network/MicroBlockSynchronizer.scala]: doesn't get cleaned if poluted from malicious node

File: src/main/scala/com/wavesplatform/network/messages.scala [hunk 2]
```diff
@@ -17,7 +17,7 @@ case class GetBlock(signature: ByteStr) extends Message
 case class LocalScoreChanged(newLocalScore: History.BlockchainScore) extends Message
 case class RawBytes(code: Byte, data: Array[Byte]) extends Message
 case class BlockForged(block: Block) extends Message
-case class MicroBlockInv(totalBlockSig: ByteStr, prevBlockSig: ByteStr)  extends Message
+case class MicroBlockInv(totalBlockSig: ByteStr, prevBlockSig: ByteStr, created: Long = System.currentTimeMillis()) extends Message
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/network/messages.scala]: No reason to use default values

File: src/main/scala/scorex/transaction/NgHistoryWriter.scala [hunk 3]
```diff
@@ -17,7 +18,7 @@ trait NgHistoryWriter extends HistoryWriter with NgHistory {
 
   def bestLiquidBlock(): Option[Block]
 
-  def forgeBlock(id: BlockId): Option[(Block, DiscardedTransactions)]
+  def forgeBlock(id: BlockId): Option[(Block, Int, DiscardedTransactions)]
```
Comment (ismagin) [inline src/main/scala/scorex/transaction/NgHistoryWriter.scala]: It's better to use `Option[(Block, DiscardedMicroblocks)]`

File: src/main/scala/com/wavesplatform/job/ParallelJobPoolAbstractDecorator.scala [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.wavesplatform.job
+
+abstract class ParallelJobPoolAbstractDecorator[ItemT, GroupIdT, JobT <: Runnable](orig: ParallelJobPool[ItemT, GroupIdT, JobT])
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/job/ParallelJobPoolAbstractDecorator.scala]: Don't forget to remove this on merge

File: src/main/scala/com/wavesplatform/network/messages.scala (hunk 2 above)
Comment (vsuharnikov) [reply src/main/scala/com/wavesplatform/network/messages.scala]: There is no reason to write `System.currentTimeMillis()` everywhere

**Comment by vsuharnikov** [comment]
The attackers should send the data a long time :) But ok, i'll fix it

File: src/main/scala/com/wavesplatform/network/MicroBlockSynchronizer.scala (hunk 1 above)
Comment (vsuharnikov) [reply src/main/scala/com/wavesplatform/network/MicroBlockSynchronizer.scala]: Ok

File: src/main/scala/scorex/transaction/NgHistoryWriter.scala (hunk 3 above)
Comment (vsuharnikov) [reply src/main/scala/scorex/transaction/NgHistoryWriter.scala]: Ok

File: src/main/scala/com/wavesplatform/job/ParallelJobPoolAbstractDecorator.scala (hunk 4 above)
Comment (vsuharnikov) [reply src/main/scala/com/wavesplatform/job/ParallelJobPoolAbstractDecorator.scala]: Done :)

**Pull request merged**